example-plugin/build.gradle

```diff
@@ -0,0 +1,70 @@
+plugins {
+    id 'com.diffplug.gradle.spotless' version '4.3.0'
+    id 'distribution'
+    // Generate IntelliJ IDEA's .idea & .iml project files
+    id 'idea'
+    id 'java'
+}
+
+repositories {
+    maven { // The google mirror is less flaky than mavenCentral()
+        url 'https://maven-central.storage-download.googleapis.com/repos/central/data/'
+    }
+    mavenLocal()
+    mavenCentral()
+}
+
+sourceCompatibility = 1.17
+targetCompatibility = 1.17
+
+group 'com.yelp.nrtsearch.plugins'
+version '0.0.1'
+
+repositories {
+    mavenCentral()
+}
+
+ext {
+    nrtsearchVersion = "0.+"
+}
+
+dependencies {
+    compileOnly "com.yelp.nrtsearch:server:${nrtsearchVersion}"
+    testImplementation "com.yelp.nrtsearch:server:${nrtsearchVersion}"
+    testImplementation("com.yelp.nrtsearch:server:${nrtsearchVersion}:tests")
+    testImplementation "junit:junit:4.13"
+    testImplementation "org.assertj:assertj-core:3.19.0"
+}
+
+distributions {
+    main {
+        contents {
+            from(jar)
+            from(this.project.configurations.runtimeClasspath)
+            from('src/main/plugin-metadata/plugin-metadata.yaml') {
```

Comment:
If there is a way to have gradle fully generate this file it might be cleaner, but that doesn't necessarily need to be solved now.